tests/unit/service/test_ansible.py

```diff
@@ -302,5 +321,252 @@ def test_run_lifecycle_keep_files(self):
+            'deployment_location': {
+                'name': 'winterfell',
+                'type': "Kubernetes",
+                'properties': PropValueMap({
+                })
+            },
+            'keep_files': True,
+            'request_id': request_id
+            })
+            
+            associated_topology = AssociatedTopology.from_dict({
+                'apache1': {
+                    'id': '12345678',
+                    'type': 'Openstack'
+                },
+                'apache2': {
+                    'id': '910111213',
+                    'type': 'Openstack'
+                }
+            })
+
+            self.assertLifecycleExecutionEqual(resp, LifecycleExecution(request_id, STATUS_COMPLETE, None, {'msg': "Associated topology returned", 'public_ip': "10.21.28.94", 'internal_ip': "10.10.10.42"}, associated_topology))
+            self.assertTrue(os.path.exists(dst))
+        finally:
+            logger.removeHandler(stream_handler)
+            
+    def ansible_missing_associated_topology_id_in_fact(self):
+        # configure so that we can see logging from the code under test
+        stream_handler = logging.StreamHandler(sys.stdout)
+        logger.addHandler(stream_handler)
+
+        try:
+            request_id = uuid.uuid4().hex
+
+            properties = PropValueMap({
+                'hello_world_private_ip': {
+                    'value': '10.220.217.113',
+                    'type': 'string'
+                },
+                'ansible_ssh_user': {
+                    'value': 'accanto',
+                    'type': 'string'
+                },
+                'ansible_ssh_pass': {
+                    'value': 'accanto',
+                    'type': 'string'
+                },
+                'ansible_become_pass': {
+                    'value': 'accanto',
+                    'type': 'string'
+                }
+            })
+            system_properties = PropValueMap({
+            })
+            
+            dst = self.__copy_directory_tree(str(pathlib.Path(__file__).parent.absolute()) + '/../../resources/ansible_missing_associated_topology_id_in_fact')
+
+            resp = self.ansible_client.run_lifecycle_playbook({
+            'lifecycle_name': 'create',
+            'driver_files': DirectoryTree(dst),
+            'system_properties': system_properties,
+            'resource_properties': properties,
+            'deployment_location': {
+                'name': 'winterfell',
+                'type': "Kubernetes",
+                'properties': PropValueMap({
+                })
+            },
+            'keep_files': True,
+            'request_id': request_id
+            })
+            
+            self.assertLifecycleExecutionEqual(resp, LifecycleExecution(request_id, STATUS_COMPLETE, None, {}, None))
+            self.assertTrue(os.path.exists(dst))
         finally:
             logger.removeHandler(stream_handler)
```

Comment:
@eflamini , I see that you've updated the code in other files to have new lines at the eof. 
Not sure, but if it is required here as well.
Ignore, if not required.